app/models/recommend_room.rb

```diff
@@ -0,0 +1,13 @@
+class RecommendRoom < ApplicationRecord
+  belongs_to :room
+  validates :room, uniqueness: true
+  validate :public_room
+
+  private
+
+    def public_room
```

Comment:
```suggestion
    def must_belongs_to_public_room
```
Railsのcustom validateメソッドはメソッド名でどんなvalidationなのか分かるように書くのが風習なのでそんな感じでお願いしたいです
http://railsdoc.com/references/validate